contrib/doc-ft-components/DEVEL.FT-REQUIREMENTS.md

```diff
@@ -0,0 +1,777 @@
+   reported. In `ompi_mtl_ofi_context_progress()`, an error may be returned from
+   `fi_cq_read()` as a consequence of a process failure; this will abort the process
+   instead of returning the error.
+
+
+The UCX PML
+-----------
+
+Similarly to OFI, UCX fault tolerance is supported using the OB1 PML, with the UCT BTL. Again, this
+is a workaround for the fact that the UCX PML does not currently support fault tolerance.
+
+1. The UCX PML does not pass flags to set UCP endpoints in fault-tolerant mode
+2. The UCX PML uses requests of type `OMPI_REQUEST_PML` that are not compatible with
+   a `ompi_pml_base_request_t`. Fields needed by `ompi_request_is_failed_fn()` are missing
+3. The UCX PML does not have a `pml_revoke_comm` function. The unexpected queues, and
+   matching logic is within the UCP component of UCX.
+4. When using the UCX PML, no BTL is initialized and the `ftagree` component and the `ompi_comm_revoke()`
+   function do direct calls to the BTL to send control messages.
+
+RMA Components
+==============
+
+The **OSC** framework is in charge of managing RMA communication.
+At the moment, fault tolerance with RMA communication is not supported.
+However, the ULFM standard draft document ***does*** define semantics for
+RMA fault tolerance, and contributions to enable this feature in Open MPI
+are welcome.
+
+I/O Components
+==============
+
+The following components manage MPI I/O in Open MPI
+
+- _fs_ File system functions for MPI I/O
+- _fbtl_ File byte transfer layer: abstraction for individual read/write
+  operations for OMPIO
+- _fcoll_ Collective read and write operations for MPI I/O
+- _sharedfp_ Shared file pointer operations for MPI I/O
+
+All components in these frameworks are unmodified, **untested** (expect clean
+post-failure abort). Again, the ULFM standard draft document **does** define
+semantics for File operation fault tolerance, and contributions to enable
+this feature in Open MPI are welcome.
+
+___________________________________________________________________________
+
+More Information
+================
+
+Author: Aurelien Bouteiller
+
+The [README.FT.ULFM.md] gives a more complete description of current
+component support in Open MPI ULFM.
+
+More information (tutorials, examples, build instructions for leading
+top500 systems) is also available in the Fault Tolerance Research
+Hub website:
+  <https://fault-tolerance.org>
+
+## Bibliographic References
+If you are looking for, or want to cite a general reference for ULFM,
+please use
+
+_Wesley Bland, Aurelien Bouteiller, Thomas Herault, George Bosilca, Jack
+J. Dongarra: Post-failure recovery of MPI communication capability: Design
+and rationale. IJHPCA 27(3): 244-254 (2013)._
+
+Available from: http://journals.sagepub.com/doi/10.1177/1094342013488238.
+___________________________________________________________________________
+
+Copyright
+=========
+
+```
+Copyright (c) 2019-2020 The University of Tennessee and The University
```

Comment:
@bosilca Did you want 2021 here?

Reply:
No, we will update the copyright next time we make changes to this file.